packages/xrpl/src/utils/parseNFTokenID.ts

```diff
@@ -0,0 +1,94 @@
+
+import { XrplError } from '../errors'
+
+/**
+ * All information encoded within an NFTokenID.
+ */
+interface NFTokenID {
+  /** The encoded hex string which represents an NFToken on ledger. */
+  TokenID: string
+  /** Which flags were enabled when the token was minted. */
+  Flags: number
+  /** The fee given to the minter on each trade of this NFToken in basis points. */
+  TransferFee: number
+  /** The original creator of this NFT. */
+  Issuer: string
+  /** A number associated with the token chosen by the minter. */
+  Taxon: number
+  /** The sequence number of the transaction which minted this token. */
+  Sequence: number
+}
+
+/**
+ * An issuer may issue several NFTs with the same taxon; to ensure that NFTs are
+ * spread across multiple pages we lightly mix the taxon up by using the sequence
+ * (which is not under the issuer's direct control) as the seed for a simple linear
+ * congruential generator.
+ *
+ * From the Hull-Dobell theorem we know that f(x)=(m*x+c) mod n will yield a
+ * permutation of [0, n) when n is a power of 2 if m is congruent to 1 mod 4 and
+ * c is odd. By doing a bitwise XOR with this permutation we can scramble/unscramble
+ * the taxon.
+ *
+ * The XLS-20d proposal fixes m = 384160001 and c = 2459.
+ * We then take the modulus of 2^32 which is 4294967296.
+ *
+ * @param taxon - The scrambled or unscrambled taxon (The XOR is both the encoding and decoding)
+ * @param tokenSeq - The account sequence when the token was minted. Used as a psuedorandom seed.
+ * @returns the opposite taxon. If the taxon was scrambled it becomes unscrambled, and vice versa.
+ */
+function unscrambleTaxon(taxon: number, tokenSeq: number): number {
+  /* eslint-disable no-bitwise -- XOR is part of the encode/decode scheme. */
+  return (taxon ^ (384160001 * tokenSeq + 2459)) % 4294967296
+  /* eslint-enable no-bitwise */
+}
+
+/**
+ * Parses an NFTokenID into the information it is encoding.
+ *
+ * Example decoding:
+ *
+ * 000B 0539 C35B55AA096BA6D87A6E6C965A6534150DC56E5E 12C5D09E 0000000C
+ * +--- +--- +--------------------------------------- +------- +-------
+ * |    |    |                                        |        |
+ * |    |    |                                        |        `---> Sequence: 12
+ * |    |    |                                        |
+ * |    |    |                                        `---> Scrambled Taxon: 314,953,886
+ * |    |    |                                              Unscrambled Taxon: 1337
+ * |    |    |
+ * |    |    `---> Issuer: rJoxBSzpXhPtAuqFmqxQtGKjA13jUJWthE
+ * |    |
+ * |    `---> TransferFee: 1337.0 bps or 13.37%
+ * |
+ * `---> Flags: 11 -> lsfBurnable, lsfOnlyXRP and lsfTransferable
+ *
+ * @param tokenID - A hex string which identifies an NFToken on the ledger.
+ * @throws XrplError when given an invalid tokenID.
+ * @returns a decoded tokenID with all information encoded within.
+ */
+function parseNFTokenID(tokenID: string): NFTokenID {
```

Comment:
Are there plans to have a method that goes in the opposite direction?

Reply:
I don't think the reverse matters, you'd need to know everything about the NFT which practically means you either minted the token, or just looked it up, and the minter can already get the NFTokenID I believe.

Reply:
The grantees I've talked to made it sound like it's kind of a pain to dig out the NFTokenID from the depths of the transaction metadata - so if we could write the reverse function (or even just a method to do the digging of the metadata) I think that would also be super helpful. IIRC the only piece of information that you don't know at the time of sending the mint tx is the sequence #.

Reply:
If you want to do that in a separate PR that's fine